Correct way of using nodemon with test runners?

When running tests using mocha or the likes, if the tests fail, and thus the runner exits with a non-zero status code, nodemon seems to think there's something broken with the process and dies without watching for changes. Is there a way to opt out of this behavior?

Here's the output, btw:

```bash
$ npm run test-dev

> funkis@0.1.4 test-dev /Users/mstade/dev/funkis
> nodemon -x "_mocha -R min $@"

8 Feb 15:57:48 - [nodemon] v1.3.6
8 Feb 15:57:48 - [nodemon] to restart at any time, enter `rs`
8 Feb 15:57:48 - [nodemon] watching: *.*
8 Feb 15:57:48 - [nodemon] starting `_mocha -R min `


  545 passing (670ms)
  2 failing

  1) Protocol when called with an empty signatures object should throw a TypeError:
     AssertionError: function (rest) { return apply(fn, this, argc, arguments) } must throw function TypeError() { [native code] }
      at Context.<anonymous> (/Users/mstade/dev/funkis/test/Protocol.js:15:39)
      at Test.Runnable.run (/Users/mstade/dev/funkis/node_modules/mocha/lib/runnable.js:221:32)
      at Runner.runTest (/Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:374:10)
      at /Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:452:12
      at next (/Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:299:14)
      at /Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:309:7
      at next (/Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:247:23)
      at Immediate._onImmediate (/Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:276:5)
      at processImmediate [as _immediateCallback] (timers.js:321:17)

  2) Protocol when called with a name, and some signatures should set immutable and enumerable properties on the protocol:
     TypeError: Signatures must be an object.
      at TypeError (native)
      at new Protocol (/Users/mstade/dev/funkis/lib/Protocol.js:8:34)
      at Protocol (/Users/mstade/dev/funkis/lib/Protocol.js:5:12)
      at Context.<anonymous> (/Users/mstade/dev/funkis/test/Protocol.js:21:15)
      at Test.Runnable.run (/Users/mstade/dev/funkis/node_modules/mocha/lib/runnable.js:221:32)
      at Runner.runTest (/Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:374:10)
      at /Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:452:12
      at next (/Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:299:14)
      at /Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:309:7
      at next (/Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:247:23)
      at Immediate._onImmediate (/Users/mstade/dev/funkis/node_modules/mocha/lib/runner.js:276:5)
      at processImmediate [as _immediateCallback] (timers.js:321:17)



util.error: Use console.error instead
[nodemon] failed to start process, possible issue with exec arguments
exception in nodemon killing node
Error: Uncaught, unspecified "error" event.
    at Error (native)
    at emit (events.js:78:13)
    at ChildProcess.<anonymous> (/Users/mstade/dev/funkis/node_modules/nodemon/lib/monitor/run.js:133:11)
    at ChildProcess.emit (events.js:101:17)
    at Process.ChildProcess._handle.onexit (child_process.js:1038:12)

----------------------------------------------------------
If appropriate, please file an error with the output from:
$ node /Users/mstade/dev/funkis/node_modules/.bin/nodemon -x _mocha -R min  --dump
At http://github.com/remy/nodemon/issues/new
----------------------------------------------------------


npm ERR! Darwin 14.1.0
npm ERR! argv "/usr/local/opt/nvm/versions/io.js/v1.0.4/bin/iojs" "/usr/local/opt/nvm/versions/io.js/v1.0.4/bin/npm" "run" "test-dev"
npm ERR! node v1.0.4
npm ERR! npm  v2.3.0
npm ERR! code ELIFECYCLE
npm ERR! funkis@0.1.4 test-dev: `nodemon -x "_mocha -R min $@"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the funkis@0.1.4 test-dev script 'nodemon -x "_mocha -R min $@"'.
npm ERR! This is most likely a problem with the funkis package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     nodemon -x "_mocha -R min $@"
npm ERR! You can get their info via:
npm ERR!     npm owner ls funkis
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/mstade/dev/funkis/npm-debug.log
```

--- a/lib/monitor/run.js
+++ b/lib/monitor/run.js
@@ -158,6 +158,22 @@ function run(options) {
       process.exit();
     }
 
+    if (code === 2 && Date.now() < config.lastStarted + 500) {
+      // something wrong with parsed command
+      utils.log.error('process failed, unhandled exit code (2)');
+      utils.log.error('');
+      utils.log.error('Either the command has a syntax error,');
+      utils.log.error('or it is exiting with reserved code 2.');
+      utils.log.error('');
+      utils.log.error('To keep nodemon running even after a code 2,');
+      utils.log.error('add this to the end of your command: || true');
+      utils.log.error('');
+      utils.log.error('nodemon will stop now so that you can fix the command.');
+      utils.log.error('');
+      bus.emit('error', code);
+      process.exit();
+    }
+
     // In case we killed the app ourselves, set the signal thusly
     if (killedAfterChange) {
       killedAfterChange = false;

--- a/lib/monitor/run.js
+++ b/lib/monitor/run.js
@@ -158,8 +158,10 @@ function run(options) {
       process.exit();
     }
 
+    // If the command failed with code 2, it may or may not be a syntax error
+    // See: http://git.io/fNOAR
+    // We will only assume a parse error, if the child failed quickly
     if (code === 2 && Date.now() < config.lastStarted + 500) {
-      // something wrong with parsed command
       utils.log.error('process failed, unhandled exit code (2)');
       utils.log.error('');
       utils.log.error('Either the command has a syntax error,');
@@ -167,6 +169,7 @@ function run(options) {
       utils.log.error('');
       utils.log.error('To keep nodemon running even after a code 2,');
       utils.log.error('add this to the end of your command: || exit 1');
+      utils.log.error('');
       utils.log.error('Read more here: https://git.io/fNOAG');
       utils.log.error('');
       utils.log.error('nodemon will stop now so that you can fix the command.');

--- a/lib/monitor/run.js
+++ b/lib/monitor/run.js
@@ -158,13 +158,6 @@ function run(options) {
       process.exit();
     }
 
-    if (code === 2) {
-      // something wrong with parsed command
-      utils.log.error('process failed, unhandled exit code (2)');
-      bus.emit('error', code);
-      process.exit();
-    }
-
     // In case we killed the app ourselves, set the signal thusly
     if (killedAfterChange) {
       killedAfterChange = false;

--- a/lib/monitor/run.js
+++ b/lib/monitor/run.js
@@ -166,7 +166,7 @@ function run(options) {
       utils.log.error('or it is exiting with reserved code 2.');
       utils.log.error('');
       utils.log.error('To keep nodemon running even after a code 2,');
-      utils.log.error('add this to the end of your command: || true');
+      utils.log.error('add this to the end of your command: || exit 1');
       utils.log.error('Read more here: https://git.io/fNOAG');
       utils.log.error('');
       utils.log.error('nodemon will stop now so that you can fix the command.');

--- a/lib/monitor/run.js
+++ b/lib/monitor/run.js
@@ -167,6 +167,7 @@ function run(options) {
       utils.log.error('');
       utils.log.error('To keep nodemon running even after a code 2,');
       utils.log.error('add this to the end of your command: || true');
+      utils.log.error('Read more here: https://git.io/fNOAG');
       utils.log.error('');
       utils.log.error('nodemon will stop now so that you can fix the command.');
       utils.log.error('');

